File: VitalSphere/VitalSphere.Services/Database/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VitalSphere.Services.Database
{
    public class Appointment
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        public int UserId { get; set; }
        public User User { get; set; } = null!;

        [Required]
        [ForeignKey(nameof(WellnessService))]
        public int WellnessServiceId { get; set; }
        public WellnessService WellnessService { get; set; } = null!;

        [Required]
        public DateTime ScheduledAt { get; set; }

        [MaxLength(500)]
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: VitalSphere/UI/vital_sphere_mobile/lib/model/appointment.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  final int id;
  final int userId;
  final String userName;
  final int wellnessServiceId;
  final String wellnessServiceName;
  final String? wellnessServiceImage;
  final DateTime scheduledAt;
  final String? notes;
  final DateTime createdAt;

  Appointment({
    this.id = 0,
    this.userId = 0,
    this.userName = '',
    this.wellnessServiceId = 0,
    this.wellnessServiceName = '',
    this.wellnessServiceImage,
    required this.scheduledAt,
    this.notes,
    required this.createdAt,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) => _$AppointmentFromJson(json);
  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}

